packages/workspace/src/node/default-workspace-server.ts

```diff
@@ -109,11 +109,20 @@ export class DefaultWorkspaceServer implements WorkspaceServer {
     private async readFromUserHome(): Promise<WorkspaceData | undefined> {
         const file = this.getUserStoragePath();
         if (await fs.pathExists(file)) {
-            const config = await fs.readJson(file);
+            let config;
+            try { // readJson throws if file is empty
+                config = await fs.readJson(file);
```

Comment:
I am wondering if we should add a function to the core package to handle the json read operations.

From what I can see, in same places `fs.readJson()` is used, and in some other places `resource.readContents()` + `jsoncparser` is used (see https://github.com/theia-ide/theia/blob/master/packages/preferences/src/browser/abstract-resource-preference-provider.ts#L80). The former one is less code, while the latter is able to handle the comments in json file.

Reply:
I would like it ! But it would seem out of the scope of this PR, so maybe this can be addressed in  a new one.